src/driver/postgres/PostgresQueryRunner.ts

```diff
@@ -2538,12 +2538,20 @@ export class PostgresQueryRunner
             .filter((column) => columnNames.indexOf(column.name) !== -1)
             .forEach((column) => (column.isPrimary = true))
 
-        const pkName = primaryColumns[0].primaryKeyConstraintName
-            ? primaryColumns[0].primaryKeyConstraintName
-            : this.connection.namingStrategy.primaryKeyName(
-                  clonedTable,
-                  columnNames,
-              )
+    
+      // primary key constraint name find fix
+        let pkName = this.connection.namingStrategy.primaryKeyName(
+            clonedTable,
+            columnNames,
+        )
+        if (primaryColumns.length > 0) {
+            pkName =
+                primaryColumns[0].primaryKeyConstraintName ||
+                this.connection.namingStrategy.primaryKeyName(
+                    clonedTable,
+                    columnNames,
+                )
+        }
```

Comment:
You can even just use the "?" operator for optional chaining like this: 
```suggestion
        const pkName = primaryColumns[0]?.primaryKeyConstraintName
            ? primaryColumns[0].primaryKeyConstraintName
            : this.connection.namingStrategy.primaryKeyName(
                  clonedTable,
                  columnNames,
              )
```